src/components/tracked-controls.js

```diff
@@ -34,6 +34,7 @@ module.exports.Component = registerComponent('tracked-controls', {
         index: data.controller,
         iterateControllerProfiles: data.iterateControllerProfiles
       });
+      el.components['tracked-controls-webxr'].updateController();
```

Comment:
I realized we're already doing this in the `play` method of the [tracked-controls-webxr](https://github.com/aframevr/aframe/blob/master/src/components/tracked-controls-webxr.js#L37). This should be redundant.

Reply:
I put a breakpoint in `play` and itwas never called

Reply:
@dmarcos play only called once when the application launches. At that point, there are no controllers since they are only available once the immersive session starts.

Reply:
Thanks for testing and the patience. `play` running once before entering immersive mode is expected and the handler for `controllersupdated` should be attached. Do you see the event being fired at all? https://github.com/aframevr/aframe/blob/master/src/components/tracked-controls-webxr.js#L40

Reply:
> Thanks for testing and the patience. `play` running once before entering immersive mode is expected 

It's called when the page loads, not when the immersive session starts. At that point, there are no controllers.
I see that the event is called when the page loads.

Reply:
Yes, the `controllersupdate` handler is attached on the play method. I expect the event to fire after entering immersive when controllers are available triggering the `controllersUpdate` method. that doesn’t seem to be happening. I wonder why

Reply:
It is called after entering the session, but at that point `update` (which registers the controllers in aframe) hasn't run yet

Reply:
update should be called on component initialization just before `play`

Reply:
yes, it's called (once for each hand) when the page loads and when the immersive session starts.

Reply:
Maybe we need to talk face to face :-)

Reply:
Proper fix would be moving the call ` this.updateController` to  an `update` method in the [tracked-controls-webxr component](https://github.com/aframevr/aframe/blob/master/src/components/tracked-controls-webxr.js)

```
play: function () {
   this.updateController();
}
```
this way if the properties change the logic to look for a matching controller will run.

Reply:
The issue comes from the `haptics` components used in moonrider that set [tracked-controls](https://github.com/supermedium/superframe/blob/master/components/haptics/index.js#L11) early. When `oculus-touch-controls`  change the properties the updateController logic doesn't run with the new properties.

Reply:
I just tried that and it fixed the problem in Moon Rider!
Do you want me to make a PR with that change?

Reply:
Awesome, thanks for testing! Yeah, incorporate the change on this PR and we can merge. We’re pretty close!